back/scripts/communities/communities_selector.py

```diff
@@ -84,15 +85,36 @@ def add_sirene_infos(self, frame: pd.DataFrame) -> pd.DataFrame:
             .rename(columns={"raison_sociale": "nom_epci"})
         )
 
+    @tracker(ulogger=LOGGER, log_start=True)
+    def add_postal_code(self, frame: pd.DataFrame) -> pd.DataFrame:
+        """
+        Adds postal code information to the communities DataFrame.
+        """
+        postal_code_df = pd.read_csv(
+            self.config["postal_code"]["url"],
+            delimiter=";",
+            encoding="latin1",
+            usecols=["#Code_commune_INSEE", "Code_postal"],
+        )
+        postal_code_df.rename(
+            columns={"#Code_commune_INSEE": "code_insee", "Code_postal": "code_postal"},
+            inplace=True,
+        )
+        postal_code_df = postal_code_df.drop_duplicates(subset=["code_insee"])
```

Comment:
nit: let's use chaining
```suggestion
        postal_code_df = pd.read_csv(
            self.config["postal_code"]["url"],
            delimiter=";",
            encoding="latin1",
            usecols=["#Code_commune_INSEE", "Code_postal"],
        ).rename(
            columns={"#Code_commune_INSEE": "code_insee", "Code_postal": "code_postal"},
            inplace=True,
        )drop_duplicates(subset=["code_insee"])
```